File: app/src/main/java/com/privat/contacts/presentation/users/model/UserItemUi.java
package com.privat.contacts.presentation.users.model;

import android.content.Context;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.function.Consumer;

public interface UserItemUi {
    int id();
    void loadAvatar(ImageView imageView);

    void showTitle(TextView textView);

    void showText(TextView textView);

    void showFavorite(ImageButton imageButton, Consumer<Integer> clickAction);
}

File: app/src/main/java/com/privat/contacts/presentation/users/model/UserItemUi.kt
package com.privat.contacts.presentation.users.model

import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import java.util.function.Consumer

interface UserItemUi {
    fun id(): Int
    fun loadAvatar(imageView: ImageView)
    fun showTitle(textView: TextView)
    fun showText(textView: TextView)
    fun showFavorite(imageButton: ImageButton, clickAction: Consumer<Int>)
}
